### PR TITLE
[Form] Incorrect timezone with DateTimeLocalizedStringTransformer

### LINKED ISSUE #19721: Incorrect date result of DateTimeLocalizedStringTransformer
https://github.com/symfony/symfony/issues/19531

After merge issue above, there is bug on the DateTimeType::class form type - DateTimeToLocalizedStringTransformer class on branch master.

My timezone is Europe/Warsaw, +2h from UTC timezone.

Part of form:

```php
$builder->add(
                'publishDateEnd',
                DateTimeType::class,
                [
                    'required' => false,
                    'widget' => 'single_text',
                    'format' => 'dd.MM.yyyy HH:mm',
                    'attr' => [
                        'placeholder' => 'dd.mm.yyyy HH:mm',
                    ],
                ]
```

When for example publishDateEnd is set as 2016-08-01 10:00:00 and I set in the web form date as 2016-08-01 12:00:00, after save form date back to 2016-08-01 10:00:00. Should be saved as 2016-08-01 12:00:00.

When I add 'view_timezone' => 'UTC' into widget option, date is display as correct date, but is invalid in database

My php.ini timezone is set as Europe/Warsaw

### PR DESCRIPTION
**Pull request by mbeccati**
| Q | A |
| --- | --- |
| Branch? | 2.7 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #19721 |
| License | MIT |
| Doc PR | n.a. |

Issue was introduced in #19541

### CODE REVIEW

**Comment by Jean85** [comment]
LGTM :+1: 

Status: reviewed

File: src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php [hunk 1]
```diff
@@ -134,6 +134,21 @@ public function testTransformWithDifferentTimezones()
...
 
+    public function testReverseTransformWithNoConstructorParameters()
+    {
+        $tz = date_default_timezone_get();
+        date_default_timezone_set('Europe/Rome');
+
+        $transformer = new DateTimeToLocalizedStringTransformer();
+
+        $this->assertEquals(
+            (new \DateTime('2010-02-03 04:05'))->format('c'),
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php]: invalid syntax on PHP 5.3 (as seen on test failures)

**Comment by nicolas-grekas** [comment]
:+1:

**Comment by jakzal** [comment]
👍

**Comment by fabpot** [comment]
Thank you @mbeccati.

**Pull request merged**